bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeChecker.java

```diff
@@ -3266,6 +3300,31 @@ private static boolean isXMLSequenceChildrenEqual(List<BXml> lhsList, List<BXml>
         return true;
     }
 
+    static boolean isRegExpType(Type targetType) {
+        if (targetType.getTag() == TypeTags.TYPE_REFERENCED_TYPE_TAG) {
+            Type referredType = ((ReferenceType) targetType).getReferredType();
+            if (referredType.getQualifiedName().equals("ballerina/lang.regexp:0:RegExp")) {
```

Comment:
Will do this or use the tag.

Reply:
Cannot use the tag because the `ballerina/lang.regexp:0:RegExp` type is a type reference type. Hence can't use `TypeUtils.getReferredType` also. Changed this such that the major version is not considered.